symengine/test_visitors.h

```diff
@@ -257,6 +257,47 @@ class PolynomialVisitor : public BaseVisitor<PolynomialVisitor>
  * are not variables will be considered to be constants.
  */
 bool is_polynomial(const Basic &b, const set_basic &variables = {});
+
+class RationalVisitor : public BaseVisitor<RationalVisitor>
+{
+private:
+    bool rational_; // are we testing for rational or irrational?
+    tribool is_rational_;
+    bool neither_ = false; // Neither rational or irrational (i.e. not real)
+
+public:
+    RationalVisitor(bool rational) : rational_{rational}
+    {
+    }
+    void bvisit(const Basic &x)
+    {
+        is_rational_ = tribool::indeterminate;
+    };
+    void bvisit(const Symbol &x)
+    {
+        is_rational_ = tribool::indeterminate;
+    };
+    void bvisit(const Number &x);
```

Comment:
Instead of this, we can do `const Rational` and `const Integer` both returning true.

Reply:
I agree and will change this.